### PR TITLE
python3.11 fixes

### PR DESCRIPTION
**Pull request by ghost**
python package builds and unittests succeed after the fix. see: https://github.com/NixOS/nixpkgs/pull/268058

longintrepr.h moved but is included by Python.h

Functions renamed in https://github.com/python/cpython/pull/31657
_PyFloat_Pack2   -> PyFloat_Pack2
_PyFloat_Pack4   -> PyFloat_Pack4
_PyFloat_Pack8   -> PyFloat_Pack8
_PyFloat_Unpack2 -> PyFloat_Unpack2
_PyFloat_Unpack4 -> PyFloat_Unpack4
_PyFloat_Unpack8 -> PyFloat_Unpack8

### CODE REVIEW

File: python/xnd/util.h [hunk 1]
```diff
@@ -36,7 +36,12 @@
 
 
 #include <Python.h>
+#if PY_VERSION_HEX < 0x030B0000
+/*
+ * Python 3.11 moved longintrepr.h and it is already included in Python.h
```
Comment (hameerabbasi) [inline python/xnd/util.h:41]: Could you also link a relevant PR/issue here?

**Review by hameerabbasi — COMMENTED** [review]
One minor change, otherwise LGTM.

File: python/xnd/util.h (hunk 1 above)
Comment (a-n-n-a-l-e-e) [reply python/xnd/util.h:41]: looks like this one: https://github.com/python/cpython/pull/28968 from browsing the repo.

Comment (hameerabbasi) [reply python/xnd/util.h:41]: I'm no expert on CPython internals. If you're satisfied, please put it into the code comment and I'll merge the PR.

Comment (a-n-n-a-l-e-e) [reply python/xnd/util.h:41]: done!

**Review by hameerabbasi — APPROVED** [review]

**Comment by hameerabbasi** [comment]
Thanks for the contribution, @a-n-n-a-l-e-e!

**Pull request merged**